how-to/workspace-platform-starter/public/common/views/frame/index.js

```diff
@@ -0,0 +1,69 @@
+let acceptableOrigin;
+
+/**
+ * Handles the message from the child frame.
+ * @param message message from child frame
+ */
+function handleMessage(message) {
+	switch (message.action) {
+		case 'update-title': {
+			document.title = message?.data?.title;
+			break;
+		}
+		case 'broadcast-context': {
+			try {
+				const context = message?.data?.context;
+				console.log('setting context on system contextual group', context);
+				fin.me.interop.setContext(context);
+			} catch (error) {
+				console.warn('You are not bound to a system context group and are unable to set context', error);
+			}
+			break;
+		}
+		default: {
+			console.warn(`Unknown action passed from frame: ${message.action}`);
+		}
+	}
+}
+
+/**
+ * Takes a post message event and handles it.
+ * @param event postMessage event that is sent from child frame
+ */
+function handleEvent(event) {
+	console.log('Received event', event);
+	console.log('Acceptable domain', acceptableOrigin);
+	const { data, origin } = event;
+	if (origin === acceptableOrigin) {
+		handleMessage(data);
+	}
+}
+
+/**
+ * Initialize the DOM.
+ */
+async function initializeDOM() {
```

Comment:
Can we move this method to the top, all the code simplifications try to make the code more chronological, to make it easier to read

Reply:
Done